src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/config/SecurityConfig.kt

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdmcertificatemanagement.config
+
+import mu.KotlinLogging
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityCustomizer
+import org.springframework.security.web.SecurityFilterChain
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher
+import org.springframework.web.cors.CorsConfiguration
+import org.springframework.web.cors.CorsConfigurationSource
+import org.springframework.web.cors.UrlBasedCorsConfigurationSource
+
+
+@EnableWebSecurity
+@ConditionalOnProperty(
+    value = ["bpdm.security.enabled"],
+    havingValue = "false",
+    matchIfMissing = true
+)
+@Configuration
+class NoAuthenticationConfig {
+
+    private val logger = KotlinLogging.logger { }
+
+    @Bean
+    fun webSecurityCustomizer(): WebSecurityCustomizer {
+        logger.info { "Disabling security for any endpoints" }
+        return WebSecurityCustomizer { web -> web.ignoring().requestMatchers(AntPathRequestMatcher("**")) }
+    }
+}
+
+@EnableWebSecurity
+@ConditionalOnProperty(
+    value = ["bpdm.security.enabled"],
```

Comment:
Same here with SpEL

Reply:
As we discussed, we cann't usie SpEL here